mlir/lib/Conversion/TosaToLinalg/TosaToLinalg.cpp

```diff
@@ -2284,8 +2284,11 @@ struct RFFT2dConverter final : public OpRewritePattern<RFFT2dOp> {
 
   static Value castIndexToFloat(OpBuilder &builder, Location loc,
                                 FloatType type, Value value) {
-    auto integerVal =
-        builder.create<arith::IndexCastUIOp>(loc, builder.getI64Type(), value);
+    auto integerVal = builder.create<arith::IndexCastUIOp>(
+        loc,
+        32 < type.getIntOrFloatBitWidth() ? builder.getI64Type()
```

Comment:
nit: it's more common to see constants as the RHS of comparison operators

Reply:
Also, should we use an integer type of the same bitwidth? For example `i16` when casting to `f16`.

Reply:
tosa.rfft2d defined only for f32, probably we should not bother unless there is some strong case for this.

Reply:
I agree with @kuhar . constant on RHS are more aligned. Could you please change?